[8.1] API documentation for Endpoint Event Filters (#1646)
elastic/security-docs: docs/management/api/_exceptions-api-reusable-content.asciidoc

//  - attribute: :endpoint-artifact-name-single-sentence-start:
//      The name of the artifact in singular form with first letter capitalized
//      Example: Trusted application
//  - attribute: :endpoint-artifact-api-doc-name:
//      Hyphenated name for the artifact api doc for section title reference/cross-links
//      Example: trusted-application
//  
//  DOCUMENT IDS:
//  The reusable content in this file references the following ids which must exist in the document:
//
//  - [#create-{endpoint-artifact-api-doc-name}-item-api]
//  - [#update-{endpoint-artifact-api-doc-name}-item-api]
//  - [#exception-{endpoint-artifact-api-doc-name}-item-entry-schema]
//

> //  Note that the section title uses the attribute :endpoint-artifact-api-doc-name for substitution

Reviewer: This needs to stay. At least that's what @paul-tavares and I agreed on [here](https://github.com/elastic/security-docs/pull/1643#r820806467)

Reviewer 2: @ashokaditya thanks for the heads up, it's added back in: https://github.com/elastic/security-docs/pull/1646/commits/9283d80e8452d295d007ac0077cf8be47b63d7c3

|==============================================
|Name |Type |Description |Required

|`comments` |xref:exception-{endpoint-artifact-api-doc-name}-item-comment-schema[comment][] a|Array of xref:exception-{endpoint-artifact-api-doc-name}-item-comment-schema[`comment`] objects. |No, defaults to empty array.
|`description` |String |Describes the {endpoint-artifact-name-single} item. |Yes
|`entries` |xref:exception-{endpoint-artifact-api-doc-name}-item-entry-schema[entry][] |Array containing the {endpoint-artifact-name-single} query conditions. See xref:exception-{endpoint-artifact-api-doc-name}-item-entry-schema[] for more details. |Yes.
|`list_id` |String a|Must be set to `pass:a[{endpoint-artifact-list-id}]`. |Yes
|`item_id` |String |Unique identifier of the exception item. |No, automatically
created when it is not provided.
|`meta` |Object |Placeholder for metadata about the exception item. |No
|`name` |String |The {endpoint-artifact-name-single} name. |Yes
|`namespace_type` |String a|Must be set to `agnostic` for endpoint artifacts. |Yes

> |`os_types` |xref:exception-{endpoint-artifact-api-doc-name}-item-os-types[os_type][] |Array of xref:exception-{endpoint-artifact-api-doc-name}-item-os-types[]. | Yes

Reviewer: I've updated `...item-os-types[]` to `item-os-types[os_types] values.` in my PR.

|==============================================
|Name |Type |Description |Required

|`comments` |xref:exception-{endpoint-artifact-api-doc-name}-item-comment-schema[comment][] a|Array of xref:exception-{endpoint-artifact-api-doc-name}-item-comment-schema[`comment`] objects. |No, defaults to empty array.
|`description` |String |Describes the {endpoint-artifact-name-single}. |Yes
|`entries` |xref:exception-{endpoint-artifact-api-doc-name}-item-entry-schema[entry][] |Array containing the {endpoint-artifact-name-single} query conditions. See xref:exception-{endpoint-artifact-api-doc-name}-item-entry-schema[] for more details. |Yes.
|`id` |String |The item's unique identifier. a|Yes, when the item's `item_id` field is not used.
|`item_id` |String a|The `item_id` of the item you are updating. a|Yes, when the item's `id` field is not used.
|`meta` |Object |Placeholder for metadata about the exception item. |No
|`name` |String |The {endpoint-artifact-name-single} name. |Yes
|`namespace_type` a|String a|Must be set to `agnostic`. |Yes

> |`os_types` |xref:exception-{endpoint-artifact-api-doc-name}-item-os-types[os_type][] |Array of xref:exception-{endpoint-artifact-api-doc-name}-item-os-types[]. | Yes

Reviewer: I've updated `...item-os-types[]` to `item-os-types[os_types] values.` in my PR.

* `linux`
* `macos`

[#exception-{endpoint-artifact-api-doc-name}-item-scope-assignment-tags]
===== Scope assignment

> {endpoint-artifact-name-sentence-start} can be assigned globally across all endpoint policies, or assigned to specific policies. The assignment of the {endpoint-artifact-name-single} is done by defining one or more tags with a prefix of `policy:`. Note that the {endpoint-artifact-name-single} can be either "global" or "per policy", but not *both*. The following tags are available for use in order to control the assignment scope:

Reviewer 3: ```suggestion
{endpoint-artifact-name-sentence-start} can be assigned globally across all endpoint policies, or assigned to specific policies. The assignment of the {endpoint-artifact-name-single} is done by defining one or more tags with a prefix of `policy:`. Note that the {endpoint-artifact-name-single} can be either global or per policy, but not *both*. The following tags are available for use in order to control the assignment scope:
```

* `linux`
* `macos`

[#exception-{endpoint-artifact-api-doc-name}-item-scope-assignment-tags]
===== Scope assignment

> {endpoint-artifact-name-sentence-start} can be assigned globally across all endpoint policies, or assigned to specific policies. The assignment of the {endpoint-artifact-name-single} is done by defining one or more tags with a prefix of `policy:`. Note that the {endpoint-artifact-name-single} can be either global or per policy, but not *both*. The following tags are available for use in order to control the assignment scope:

Reviewer 4: ```suggestion
{endpoint-artifact-name-sentence-start} can be assigned globally across all endpoint policies, or assigned to specific policies. You can assign the {endpoint-artifact-name-single} by defining one or more tags with a prefix of `policy:`. Note that the {endpoint-artifact-name-single} can be either global or per policy, but not *both*. The following tags are available for use in order to control the assignment scope:
```
